tests/components/evohome/test_storage.py

```diff
@@ -0,0 +1,140 @@
+"""The tests for evohome storage load & save."""
+
+from typing import Any, Final
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components.evohome import (
+    ACCESS_TOKEN_EXPIRES,
+    CONF_USERNAME,
+    DOMAIN,
+    SZ_SESSION_ID,
+    USER_DATA,
+    EvoSession,
+    dt_aware_to_naive,
+)
+from homeassistant.core import HomeAssistant
+import homeassistant.util.dt as dt_util
+
+DIFF_EMAIL_ADDRESS: Final = "diff_user@email.com"
+SAME_EMAIL_ADDRESS: Final = "same_user@email.com"
+
+DATE_TIME_EXPIRES: Final = "2024-06-10T22:05:54+00:00"  # tests need UTC TZ
+
+
+TEST_DATA: Final = {
+    "00": None,
+    "01": {},
+    "10": {
+        "username": SAME_EMAIL_ADDRESS,
+        "refresh_token": "jg68ZCKYdxEI3fF...",
+        "access_token": "1dc7z657UKzbhKA...",
+        "access_token_expires": DATE_TIME_EXPIRES,
+    },  # no need to test "user_data": {"sessionId": None},
+    "11": {
+        "username": SAME_EMAIL_ADDRESS,
+        "refresh_token": "jg68ZCKYdxEI3fF...",
+        "access_token": "1dc7z657UKzbhKA...",
+        "access_token_expires": DATE_TIME_EXPIRES,
+        "user_data": {"sessionId": "F7181186..."},
+    },
+}
+
+
+@pytest.mark.parametrize("idx", TEST_DATA)
+async def test_load_auth_tokens_same(
+    hass: HomeAssistant, hass_storage: dict[str, Any], idx: str
+) -> None:
+    """Test restoring authentication tokens when matching account."""
+
+    hass_storage[DOMAIN] = {
+        "version": 1,
+        "minor_version": 1,
+        "key": "evohome",
+        "data": TEST_DATA[idx],
+    }
+
+    # Load access tokens and session_id from the store
+    sess = EvoSession(hass)
```

Comment:
Don't interact with integration details. Patch the library and assert things about the library that uses the tokens. Set up the integration to load the tokens.